ocaml/xapi/repository.ml

```diff
@@ -512,50 +512,53 @@ let get_host_updates_in_json ~__context ~installed =
         (ExnHelper.string_of_exn e) ;
       raise Api_errors.(Server_error (get_host_updates_failed, [ref]))
 
+(* Map the URI in the HTTP reuqest to local repo path *)
+let map_uri_to_path ~prefix ~by uri =
+  let regex = prefix |> Re.str |> Re.compile in
+  uri
+  |> Uri.of_string
+  |> Uri.path
+  |> Re.replace_string ~all:false regex ~by
+  |> Helpers.resolve_uri_path ~root:!Xapi_globs.local_pool_repo_dir
+
 let get_repository_handler (req : Http.Request.t) s _ =
   let open Http in
-  let open Xapi_stdext_std.Xstringext in
-  debug "Repository.get_repository_handler URL %s" req.Request.uri ;
+  debug "%s URL: %s" __FUNCTION__ req.Request.uri ;
   req.Request.close <- true ;
   if Fileserver.access_forbidden req s then
     Http_svr.response_forbidden ~req s
   else
-    let can_be_authorized =
-      try
-        Xapi_http.with_context "get_repository_handler" req s (fun _ -> ()) ;
```

Comment:
This does a session check. If we remove it here, then we need to make sure that authentication is checked right at the beginning of the HTTP handler, in `Xapi_http.add_handler`, by removing `"get_repository"` from `public_http_actions_with_no_rbac_check` in datamodel.ml.